frontends/p4/def_use.cpp

```diff
@@ -315,7 +315,7 @@ Definitions *Definitions::joinDefinitions(const Definitions *other) const {
         auto loc = d.first;
         auto defs = d.second;
         auto current = ::P4::get(definitions, loc);
-        if (current != nullptr) {
+        if (current != nullptr && current != defs) {
```

Comment:
Might make more sense to put this check in the `merge` method.  Could even change that to not create a new ProgramPoints object until it finds a point not in the merged-to set:

~~~
const ProgramPoints *ProgramPoints::merge(const ProgramPoints *with) const {
    auto *result = nullptr;
    for (auto *pp : with->points) {
        if (!points.count(pp) {
            if (!result) result = new ProgramPoints(points);
            result->points.insert(pp);
        }
    }
    return result ? result : this;
}
~~~

Reply:
@ChrisDodd: Thanks for taking a look.  I've moved the identity-check into `merge()`, and have added your suggested allocation-deferral logic.  I've also changed `merge()` to iterate over the smaller of the two `ProgramPoint`s, thus detecting when one is a subset of the other in either order.